src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -0,0 +1,1790 @@
+                //         (C1 a, (C2 b, C3 c)) = ("hello", ("dog", "cat"));
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""cat""").WithArguments("string", "C3").WithLocation(7, 50)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void ExpressionTree_01()
+        {
+            var source = @"
+using System;
+using System.Linq.Expressions;
+class C
+{
+    static void Main()
+    {
+        Expression<Func<byte[]>> x = () => ""hello"";
+        Expression<FuncSpanOfByte> y = () => ""dog"";
+        Expression<FuncReadOnlySpanOfByte> z = () => ""cat"";
+
+        System.Console.WriteLine(x);
+        System.Console.WriteLine(y);
+        System.Console.WriteLine(z);
+    }
+
+    delegate Span<byte> FuncSpanOfByte();
+    delegate ReadOnlySpan<byte> FuncReadOnlySpanOfByte();
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            // PROTOTYPE(UTF8StringLiterals) : Confirm this is the behavior we want. Or should we disallow this conversion in expression tree?
+            CompileAndVerify(comp, expectedOutput: @"
+() => new [] {104, 101, 108, 108, 111}
+() => new Span`1(new [] {100, 111, 103})
+() => new ReadOnlySpan`1(new [] {99, 97, 116})
+").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void ExpressionTree_02()
+        {
+            var source = @"
+using System;
+using System.Linq.Expressions;
+class C
+{
+    static void Main()
+    {
+        Expression<Func<C1>> x = () => (C1)""hello"";
+        Expression<Func<C2>> y = () => (C2)""dog"";
+        Expression<Func<C3>> z = () => (C3)""cat"";
+
+        System.Console.WriteLine(x);
+        System.Console.WriteLine(y);
+        System.Console.WriteLine(z);
+    }
+}
+
+class C1
+{
+    public static implicit operator C1(byte[] x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static implicit operator C2(Span<byte> x)
+    {
+        return new C2();
+    }
+}
+
+class C3
+{
+    public static implicit operator C3(ReadOnlySpan<byte> x)
+    {
+        return new C3();
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            // PROTOTYPE(UTF8StringLiterals) : Confirm this is the behavior we want. Or should we disallow this conversion in expression tree?
+            CompileAndVerify(comp, expectedOutput: @"
+() => Convert(new [] {104, 101, 108, 108, 111}, C1)
+() => Convert(new Span`1(new [] {100, 111, 103}), C2)
+() => Convert(new ReadOnlySpan`1(new [] {99, 97, 116}), C3)
+").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_01()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
```

Comment:
>Test(""s"") [](http://example.com/codeflow?start=33&length=11)

Why do we prefer `ReadOnlySpan<char>` over `byte[]` rather than being ambiguous?

Reply:
> Why do we prefer ```ReadOnlySpan<char>``` over ```byte[]``` rather than being ambiguous?

Conversion to ```ReadOnlySpan<char>``` is not a new conversion.